magma/interface.py

```diff
@@ -379,7 +379,11 @@ def __init__(self, **kwargs):
         self.__decl = []
         for name, typ in kwargs.items():
             ref = LazyDefnRef(name=name)
-            port = typ.flip()(name=ref)
+            if isinstance(typ, MagmaProtocolMeta):
+                port = typ.flip()
+                port = port._from_magma_value_(port._to_magma_()(name=ref))
```

Comment:
why do we need to lower and lift here? think I'm missing something. We could make this part of the metacls `__call__` so we don't have to do this everytime we want to instance a value of it.

Reply:
Because the containing type doesn't necessary need to have an interface where it can be constructed with a magma Ref (name).  I discussed with caleb, the options were to have either an enforced interface where a type can be constructed with a ref or to have an interface that constructs an instance from a value constructed with a ref (magma types must satisfy the interface to construct with a ref).  This was more explicit so that's what we chose.  If we want to change that, it's really an interface level change we'd want to discuss with @cdonovick

Reply:
More discussion here: https://github.com/phanrahan/magma/pull/566